runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -310,12 +343,69 @@ void chpl_mem_layerInit(void) {
   //   jemalloc 4.5.0 man: "Once, when the first call is made to one of the
   //   memory allocation routines, the allocator initializes its internals"
   if (heap_base != NULL) {
-    heap.base = heap_base;
-    heap.size = heap_size;
-    heap.cur_offset = 0;
-    if (pthread_mutex_init(&heap.alloc_lock, NULL) != 0) {
-      chpl_internal_error("cannot init chunk_alloc lock");
+    int hpi;
```

Comment:
I think this logic should be stuffed into a helper routine that's called from initializeSharedHeap()